src/mlpack/methods/ann/loss_functions/poisson_nll_loss.hpp

```diff
@@ -124,6 +124,18 @@ class PoissonNLLLoss
   void serialize(Archive& ar, const unsigned int /* version */);
 
  private:
+  //! Check if the probabilities lie in the range [0, 1].
+  template<typename eT>
+  void CheckProbs(const arma::Mat<eT>& probs)
+  {
+    for (size_t i = 0; i < probs.size(); ++i)
+    {
+      if (probs[i] > 1.0 || probs[i] < 0.0)
```

Comment:
Why does each input should be <= 1?

Reply:
Because when `logInput = false`, it is expected that the input will be probabilities (for example output of a softmax layer) and 0 <= p <= 1.